util/syncpacket/src/lib.rs

```diff
@@ -0,0 +1,144 @@
+// Copyright 2015-2019 Parity Technologies (UK) Ltd.
+// This file is part of Parity Ethereum.
+
+// Parity Ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Ethereum.  If not, see <http://www.gnu.org/licenses/>.
+
+#![recursion_limit="128"]
+
+// Needs to be "extern crate" even in rust 2018:
+// https://blog.rust-lang.org/2018/12/21/Procedural-Macros-in-Rust-2018.html
+extern crate proc_macro;
+
+use proc_macro2::Ident;
+use quote::quote;
+use syn::{Data, DeriveInput, Meta, MetaList, NestedMeta};
+
+fn parse_protocol_arguments(args: MetaList) -> Ident {
+	if args.nested.len() != 1 {
+		panic!("protocol attribute should have exactly one argument");
```

Comment:
Can we turn those into compiler errors? See: https://github.com/paritytech/jsonrpc/pull/382/files

Reply:
Sure, I didn't know about that, thanks for the pointer.